lisa/util/__init__.py

```diff
@@ -231,3 +234,22 @@ def is_valid_url(url: str, raise_error: bool = True) -> bool:
 
 def filter_ansi_escape(content: str) -> str:
     return __ansi_escape.sub("", content)
+
+
+def dump_file(file_name: Path, content: Any) -> None:
+    local_path_parts = constants.RUN_LOCAL_PATH.parts
+    parts_len = len(local_path_parts)
+    run_path = Path(Path()/local_path_parts[parts_len-5] /
+        local_path_parts[parts_len-4] / local_path_parts[parts_len-3] /
+        local_path_parts[parts_len-2] / local_path_parts[parts_len-1])
+    if(file_name.is_absolute()):
+        if(not str(file_name).find(str(run_path))):
+            raise error
+    else:  # file_name is a relative path
+        abs_path = Path.cwd()/file_name
+        if(not str(abs_path).find(str(run_path))):
+            raise error
+
+    file = open(file_name, "w")
```

Comment:
We can make sure the path exists before open a file. It would be more convenient.

`file_name.parent.mkdir(parents=True, exist_ok=True)`

Reply:
use `with` keyword is more safe to recycle resource, and reduce code lines.

```python
with open(file_name, "w") as f:
    f.write...
```

Reply:
The second comments is not resolved.

> use `with` keyword is more safe to recycle resource, and reduce code lines.
> 
> ```python
> with open(file_name, "w") as f:
>     f.write...
> ```